src/ESLint.Formatter/sarif.js

```diff
@@ -132,9 +132,11 @@ module.exports = function (results, data) {
                     console.log(err);
                 }
             }
-            if (result.messages.length > 0) {
 
-                result.messages.forEach(message => {
+            const messages = result.messages.concat(result.suppressedMessages);
```

Comment:
Is `suppressedMessages` guaranteed to be not `undefined`?

Reply:
Yes. The official doc is [here](https://eslint.org/docs/developer-guide/nodejs-api#-lintresult-type).

[_distinguishSuppressedMessages()](https://github.com/eslint/eslint/blob/6f940c3ce715327f282c197d0f71b91848e5d83d/lib/linter/linter.js#L1830) in [linter](https://github.com/eslint/eslint/blob/main/lib/linter/linter.js) is to preserve the latest suppressed message list, which is initially an empty list instead of `undefined` or `null`. This [comment](https://github.com/eslint/eslint/pull/15459#discussion_r790648587) could also corroborate this.

Reply:
Thanks @Yiwei-Ding. Any backwards compatibility concerns? Would our formatter ever receive output from an earlier version of ESLint that does not output `suppressedMessages` property?

Everything else looks great. Thanks for the contribution.

Reply:
Thanks for reminding! It's good for me to add incompatibility check.

```js
let messages = result.messages;

if (result.suppressedMessages) {
    messages = messages.concat(result.suppressedMessages);
}

...
```

Reply:
Thanks for addressing that!